packages/components/src/filter-picker/test/index.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * External dependencies
+ */
+import { render } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+/**
+ * Internal dependencies
+ */
+import { Basic } from '../stories/index';
+import FilterPicker from '../index';
+import Search from '../../search';
+import productAutocompleter from '../../search/autocompleters/product';
+// Due to Jest implementation we cannot mock it only for specific tests.
+// If your test requires non-mocked Search, move them to another test file.
+jest.mock( '../../search' );
+
+describe( 'FilterPicker', () => {
+	it( 'should render the example from the storybook', async () => {
+		// Jest and its JSDOM does not allow making extensive use of searchParams used by Basic example.
+		const path = '/story/woocommerce-admin-components-filterpicker--basic';
+
+		expect( function () {
+			render( <Basic path={ path } /> );
+		} ).not.toThrow();
+	} );
+	describe( "when a config is given with a filter with `component: 'Search'`", () => {
+		let config;
+		beforeEach( () => {
+			config = {
+				label: 'Show',
+				staticParams: [],
+				param: 'product_filter',
+				showFilters: () => true,
+				filters: [
+					{ label: 'All Products', value: 'all' },
+					{
+						component: 'Search',
+						value: 'select_product',
+						chartMode: 'item-comparison',
+						path: 'select_product',
+						settings: {
+							type: 'products',
+							param: 'products',
+							labels: {
+								placeholder: 'Type to search for a product',
+								button: 'Single Product',
+							},
+						},
+					},
+				],
+			};
+		} );
+
+		it( 'should render the Search component', async () => {
+			const path = '/foo/bar';
+
+			const { queryAllByRole } = render(
+				<FilterPicker path={ path } type="custom" config={ config } />
```

Comment:
@tomalec , are we sure about having `type="custom"` here? From what I see in the FilterPicker code, the type "custom" should be specified inside filter settings and not `FilterPicker` props?

Reply:
You're right, thanks. Removed in https://github.com/woocommerce/woocommerce-admin/pull/6880/commits/98c256afe9445f2cea15813b4b35aaac9204d563